packages/jsii-rosetta/lib/tablets/tablets.ts

```diff
@@ -67,60 +79,43 @@ export class LanguageTablet {
 
   private toSchema(): TabletSchema {
     return {
-      version: '1',
+      version: CURRENT_SCHEMA_VERSION,
       toolVersion: TOOL_VERSION,
-      snippets: mapValues(this.snippets, (s) => s.toSchema()),
+      snippets: mapValues(this.snippets, (s) => s.snippet),
     };
   }
 }
 
+/**
+ * Mutable operations on an underlying TranslatedSnippetSchema
+ */
 export class TranslatedSnippet {
   public static fromSchema(schema: TranslatedSnippetSchema) {
-    const ret = new TranslatedSnippet();
-    Object.assign(ret.translations, schema.translations);
-    Object.assign(ret._fqnsReferenced, schema.fqnsReferenced);
-    Object.assign(ret._syntaxKindCounter, schema.syntaxKindCounter);
-    ret._didCompile = schema.didCompile;
-    ret._where = schema.where;
-    ret.fullSource = schema.fullSource;
-    return ret;
+    if (!schema.translations[ORIGINAL_SNIPPET_KEY]) {
+      throw new Error(`Input schema must have '${ORIGINAL_SNIPPET_KEY}' key set in translations`);
+    }
+    return new TranslatedSnippet(schema);
   }
 
-  public static fromSnippet(original: TypeScriptSnippet, didCompile?: boolean) {
-    const ret = new TranslatedSnippet();
-    Object.assign(ret.translations, {
-      [ORIGINAL_SNIPPET_KEY]: { source: original.visibleSource },
+  public static fromTypeScript(original: TypeScriptSnippet, didCompile?: boolean) {
+    return new TranslatedSnippet({
+      translations: {
+        [ORIGINAL_SNIPPET_KEY]: { source: original.visibleSource, version: '0' },
+      },
+      didCompile: didCompile,
+      location: original.location,
+      fullSource: original.completeSource,
     });
-    ret._didCompile = didCompile;
-    ret._where = original.where;
-    ret.fullSource = original.completeSource;
-    return ret;
   }
 
-  private readonly translations: Record<string, TranslationSchema> = {};
-  private readonly _fqnsReferenced = new Array<string>();
-  private readonly _syntaxKindCounter: Record<string, number> = {};
-  private _key?: string;
-  private _didCompile?: boolean;
-  private _where = '';
-  private fullSource?: string;
-
-  private constructor() {}
-
-  public get didCompile() {
-    return this._didCompile;
-  }
+  public readonly snippet: TranslatedSnippetSchema;
```

Comment:
`~` The naming here is a bit odd. `TranslatedSnippet.snippet`, which returns a `TranslatedSnippetSchema`, which reads to me like the snippet itself. It leads to a lot of `snippet.snippet` in the code, which is never great...

It's _almost_ like `TranslatedSnippet` should implement `TranslatedSnippetSchema`. Am I off there?

Reply:
Maybe the names would be better to say `TranslatedSnippetOperations` and `TranslatedSnippet`. The class represents the operations on the pure-data object.

But I do need the ability to get the pure-data object out in a lightweight way.